File: ruby-pr/53/ruby/code-quality-readability-maintainability/data_validator.rb
class DataValidator
  def validate_user(name, email, age, phone)
    # {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
    return false if name.nil? || name.length < 2 || name.length > 50 || email.nil? || !email.include?('@') || email.length < 5 || age < 18 || age > 120 || phone.nil? || phone.length != 10
    true
    # {/fact}
  end
end

File: kotlin-pr/50/kotlin/code-quality-readability-maintainability/DataValidator.kt
class DataValidator {
    fun validateUser(name: String?, email: String?, age: Int, phone: String?): Boolean {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
        return !(name == null || name.length < 2 || name.length > 50 || email == null || !email.contains("@") || email.length < 5 || age < 18 || age > 120 || phone == null || phone.length != 10)
        // {/fact}
    }
}